pyiron_contrib/tinybase/lammps.py

```diff
@@ -0,0 +1,189 @@
+import os
+from tempfile import TemporaryDirectory
+
+from ase import Atoms
+from pymatgen.io.lammps.outputs import (
+        parse_lammps_dumps,
+        parse_lammps_log,
+)
+
+from pyiron_atomistics.lammps.potential import (
+        LammpsPotential,
+        LammpsPotentialFile,
+        list_potentials
+)
+from pyiron_atomistics.lammps.control import LammpsControl
+
+from pyiron_contrib.tinybase.container import (
+        AbstractInput,
+        AbstractOutput,
+        StorageAttribute,
+        StructureInput,
+        EnergyPotOutput,
+        EnergyKinOutput,
+        ForceOutput,
+        MDOutput,
+)
+from pyiron_contrib.tinybase.task import (
+        AbstractTask,
+        ReturnStatus,
+)
+from pyiron_contrib.tinybase.shell import (
+        ShellTask,
+        ExecutablePathResolver
+)
+
+
+class LammpsInputInput(AbstractInput):
+    working_directory = StorageAttribute().type(str)
+
+    calc_type = StorageAttribute()
+
+    def calc_static(self):
+        self.calc_type = 'static'
+
+    def check_ready(self):
+        return self.working_directory is not None \
+                and self.calc_type == "static" \
+                and super().check_ready()
+
+class LammpsInputOutput(AbstractOutput):
+    working_directory = StorageAttribute().type(str)
+
+class LammpsInputTask(AbstractTask):
+    """
+    Write a set of input files for lammps calculations.
+
+    The potential input is written to `potential.inp` together with any
+    additional files needed by the potential.
+
+    The structure is written to `structure.inp`.
+
+    The lammps input script is to `control.inp`.
+    """
+
+    def _get_input(self):
+        return LammpsInputInput()
+
+    def _get_output(self):
+        return LammpsInputOutput()
+
+    def _execute(self, output):
+        with open(
+                os.path.join(self.input.working_directory, "structure.inp"),
+                "w"
+        ) as f:
+            self.input.structure.write(f, format="lammps-data")
+
+        potential = LammpsPotential()
+        potential.df = LammpsPotentialFile().find_by_name(self.input.potential)
+        potential.write_file(
+            file_name="potential.inp", cwd=self.input.working_directory
+        )
+        potential.copy_pot_files(self.input.working_directory)
+
+        control = LammpsControl()
+        if self.input.calc_type == "static":
+            control.calc_static()
+        else:
+            assert "Cannot happen"
```

Comment:
```suggestion
            assert False, "Cannot happen"
```

Reply:
From what I know, [assert is not compatible with PEP8](https://stackoverflow.com/questions/16065482/how-to-format-a-python-assert-statement-that-complies-with-pep8)

Reply:
I guess what makes sense is:

```python
assert self.input.calc_type == "static"
control.calc_static()
```

Otherwise `assert False` looks super clumsy...

Reply:
Fair enough, I'll change that.

Reply:
> From what I know, [assert is not compatible with PEP8](https://stackoverflow.com/questions/16065482/how-to-format-a-python-assert-statement-that-complies-with-pep8)

Not sure I follow, the link just says it's difficult to fit assert statements within 80 chars (which we don't adhere to anyway), but also gives solutions.

Reply:
yeah to be honest I also never understood. @sudarsan-surendralal pointed it out some years ago because apparently some tests failed because of `assert`, but I personally also like it and use it in multiple places.